### PR TITLE
Changed action for getting files and folders. Combined actions into a…

### PR DESCRIPTION
**Pull request by caseychoiniere**
… singular action

### CODE REVIEW

File: app/scripts/pages/app.jsx [hunk 1]
```diff
@@ -97,7 +97,7 @@ class App extends React.Component {
             <span>
                 <div className="statusbar-overlay"></div>
                 <div className="panel-overlay"></div>
-                {!this.state.appConfig.apiToken ? '' : <LeftMenu {...this.props} {...this.state} />}
+                {!this.state.appConfig.apiToken ? '' : <LeftMenu />}
```
Comment (dburdick) [inline app/scripts/pages/app.jsx:100]: You don't need the app's state/props (i.e. MainStore) in the LeftMenu?

File: app/scripts/actions/projectActions.js [hunk 2]
```diff
@@ -314,25 +311,8 @@ ProjectActions.editFile.preEmit = function (id, fileName) {
...
-    })
-        .catch(function (ex) {
-            ProjectActions.getContainerError(ex)
-        });
-};
-
-ProjectActions.getFileContainer.preEmit = (id) => {
-    fetch(urlGen.routes.ddsUrl + 'files/' + id, {
+ProjectActions.getEntity.preEmit = (id, kind) => {
+    fetch(urlGen.routes.ddsUrl + kind + id, {
```
Comment (dburdick) [inline app/scripts/actions/projectActions.js:315]: you should probably concatenate in the '/' between kind and id and not have your action emitters make a new type of kind (i.e. 'folder' vs 'folder/' as kinds)

File: app/scripts/pages/file.jsx [hunk 3]
```diff
@@ -17,17 +17,29 @@ class File extends React.Component {
...
     componentDidMount() {
+        let kind = 'files/';
         let id = this.props.params.id;
         this.unsubscribe = ProjectStore.listen(state => this.setState(state));
-        ProjectActions.getFileContainer(id);
+        this._loadFile(id, kind);
+    }
+
+    componentDidUpdate(prevProps) {
+        let kind = 'files/';
```
Comment (dburdick) [inline app/scripts/pages/file.jsx:27]: convert to 'files'

File: app/scripts/pages/folder.jsx [hunk 4]
```diff
@@ -16,15 +16,17 @@ class Folder extends React.Component {
     }
 
     componentDidMount() {
+        let kind = 'folders/';
```
Comment (dburdick) [inline app/scripts/pages/folder.jsx:19]: 'folders'

File: app/scripts/stores/projectStore.js [hunk 5]
```diff
@@ -282,7 +283,8 @@ var ProjectStore = Reflux.createStore({
     },
 
     editFileSuccess(id) {
-        ProjectActions.getFileContainer(id);
+        let kind = 'files/';
```
Comment (dburdick) [inline app/scripts/stores/projectStore.js:286]: 'files'

File: app/scripts/actions/projectActions.js (hunk 2 above)
Comment (caseychoiniere) [inline app/scripts/actions/projectActions.js:315]: Kind in the API is dds-folder and dds-file, so this isn't intended to follow the standards set by the api. It's simply for use in this one situation. I will concatenate in the '/' though.

**Pull request merged**